cmake/std/atdm/contributed/blake/environment.sh

```diff
@@ -0,0 +1,212 @@
+  export OMPI_FC=`which ifort`
+  export ATDM_CONFIG_LAPACK_LIBS="${LOCAL_LAPACK_LIBRARIES}"
+  export ATDM_CONFIG_BLAS_LIBS="${LOCAL_BLAS_LIBRARIES}"
+  export ATDM_CONFIG_MPI_POST_FLAGS="-bind-to;socket;-map-by;socket"
+elif [[ "$ATDM_CONFIG_COMPILER" == "GNU-7.2.0" ]] ; then
+  module load devpack/20171203/openmpi/2.1.2/gcc/7.2.0
+  module swap cmake/3.9.0 cmake/3.12.3
+  module load ninja/1.7.2
+  module load python/2.7.13
+
+  #export LOCAL_BLAS_LIBRARIES="-mkl;${MKLROOT}/lib/intel64/libmkl_intel_lp64.a;${MKLROOT}/lib/intel64/libmkl_intel_thread.a;${MKLROOT}/lib/intel64/libmkl_core.a"
+  #export LOCAL_LAPACK_LIBRARIES=${LOCAL_BLAS_LIBRARIES}
+
+  export ATDM_CONFIG_CXX_FLAGS="-g"
+  export ATDM_CONFIG_C_FLAGS="-g"
+  export ATDM_CONFIG_Fortran_FLAGS="-g"
+  #export ATDM_CONFIG_EXTRA_LINK_FLAGS=""
+  #export ATDM_CONFIG_CXX_FLAGS="-D_GLIBCXX_USE_CXX11_ABI=0"
+  export OMPI_CXX=`which g++`
+  export OMPI_CC=`which gcc`
+  export OMPI_FC=`which gfortran`
+  export ATDM_CONFIG_LAPACK_LIBS="-L${LAPACK_ROOT}/lib;-llapack;-lgfortran;-lgomp;-lm"
+  export ATDM_CONFIG_BLAS_LIBS="-L${BLAS_ROOT}/lib;-lblas;-lgfortran;-lgomp;-lm"
+#  export ATDM_CONFIG_LAPACK_LIBS="${LAPACK_ROOT}/lib/libopenblas.a"
+#  export ATDM_CONFIG_BLAS_LIBS="${BLAS_ROOT}/lib/libopenblas.a"
+  export ATDM_CONFIG_MPI_POST_FLAGS="-bind-to;socket;-map-by;socket"
+else
+  echo
+  echo "***"
+  echo "*** ERROR: COMPILER=$ATDM_CONFIG_COMPILER is not supported!"
+  echo "***"
+  return
+fi
+
+if [[ "$ATDM_CONFIG_COMPILER" == "CUDA"* ]] && \
+  [[ "${ATDM_CONFIG_COMPLEX}" == "ON" ]] && \
+  [[ "${ATDM_CONFIG_SHARED_LIBS}" == "ON" ]] ; then
+  export ATDM_CONFIG_USE_NINJA=OFF
+  export ATDM_CONFIG_CMAKE_CXX_USE_RESPONSE_FILE_FOR_OBJECTS=OFF
+fi
+# NOTE: The reason for the above logic is that 'nvcc' can't handle *.rsp
+# response files that CMake switches to when the command-lines become too long
+# and there is no way to turn off the usage of response files with the CMake
+# Ninja generator as of CMake 3.14.0.  The problem is that when CUDA and
+# complex are enabled and shared libs are used, 'nvcc' is used to create the
+# kokkoskernels shared lib which has a ton of object files which triggers
+# internal CMake logic to condense these down into a single kokkoskernels.rsp
+# file that it passes to 'nvcc' to create the kokkoskernels library.  When the
+# CMake-generated *.rsp file is passed to 'nvcc', it does not know how to
+# process it and it gives the error "No input files specified".  The
+# workaround is to switch to the CMake Makefile generator which allows and
+# turning off the usage of response files for the list of object files (and
+# 'nvcc' seems to be able to handle this long command-line in this case).
+# Note that we don't yet need to switch to use the CMake Makefile generator
+# for 'static' builds since 'ar' is used to create the kokkoskernels lib which
+# does not seem to have a problem with the creation of this lib.  (See
+# TRIL-255 and TRIL-264.)
```

Comment:
Is this still true?  Just asking.

Reply:
Oop, this is copy-paste, I'll delete; I'm not sure about the current status of that note on the relevant machines.

Reply:
@bartlettroscoe I deleted the comment (and logic above it), but my commit dismissed your review, can you approve one more time please?